presto-native-execution/presto_cpp/main/http/tests/HttpJwtTest.cpp

```diff
@@ -0,0 +1,227 @@
+  folly::init(&argc, &argv, true);
+  return RUN_ALL_TESTS();
+}
+
+class HttpJwtTestSuite : public ::testing::TestWithParam<bool> {};
+
+TEST_P(HttpJwtTestSuite, basicJwtTest) {
+  auto memoryPool = defaultMemoryManager().addLeafPool("basicJwtTest");
+
+  const std::string kSecret{"mysecret"};
+  const std::string kNodeId{"testnode"};
+
+  std::unordered_map<std::string, std::string> systemValues{
+      {std::string(SystemConfig::kMutableConfig), std::string("true")},
+      {std::string(SystemConfig::kInternalCommunicationJwtEnabled),
+       std::string("true")},
+      {std::string(SystemConfig::kInternalCommunicationSharedSecret), kSecret}};
+  auto rawSystemConfig = std::make_unique<core::MemConfigMutable>(systemValues);
+
+  auto systemConfig = SystemConfig::instance();
+  systemConfig->initialize(std::move(rawSystemConfig));
+
+  std::unordered_map<std::string, std::string> nodeValues{
+      {std::string(NodeConfig::kMutableConfig), std::string("true")},
+      {std::string(NodeConfig::kNodeId), std::string(kNodeId)}};
+  auto rawNodeConfig = std::make_unique<core::MemConfigMutable>(nodeValues);
+  auto nodeConfig = NodeConfig::instance();
+  nodeConfig->initialize(std::move(rawNodeConfig));
+
+  EXPECT_EQ(systemConfig->internalCommunicationJwtEnabled(), true);
+  EXPECT_EQ(systemConfig->internalCommunicationSharedSecret(), kSecret);
+  EXPECT_EQ(nodeConfig->nodeId(), kNodeId);
+
+  const bool useHttps = GetParam();
+
+  auto server = getServer(useHttps);
+
+  server->registerGet("/ping", ping);
+
+  std::vector<std::unique_ptr<proxygen::RequestHandlerFactory>> filters;
+  filters.push_back(
+      std::make_unique<http::filters::InternalAuthenticationFilterFactory>());
+  auto wrapper = std::make_unique<HttpServerWrapper>(std::move(server));
+  wrapper->setFilters(filters);
+
+  auto serverAddress = wrapper->start().get();
+
+  HttpClientFactory clientFactory;
+  auto client = clientFactory.newClient(
+      serverAddress, std::chrono::milliseconds(1'000), useHttps, memoryPool);
+
+  auto response = sendGet(client.get(), "/ping").get();
+  ASSERT_EQ(response->headers()->getStatusCode(), http::kHttpOk);
+
+  wrapper->stop();
+}
+
+std::unique_ptr<http::HttpResponse> jwtTestWrapper(
+    std::shared_ptr<MemoryPool> memoryPool,
+    const bool useHttps,
+    std::unordered_map<std::string, std::string>& initalSystemConfig,
+    std::unordered_map<std::string, std::string>& changedSystemConfig,
+    const uint64_t sendDelayMs = 0) {
+  const std::string kSecret{"mysecret"};
+  // Set up common system config.
+  initalSystemConfig.emplace(
+      std::string(SystemConfig::kMutableConfig), std::string("true"));
+  initalSystemConfig.emplace(
+      std::string(SystemConfig::kInternalCommunicationSharedSecret), kSecret);
+
+  auto rawSystemConfig =
+      std::make_unique<core::MemConfigMutable>(initalSystemConfig);
+  auto systemConfig = SystemConfig::instance();
+  systemConfig->initialize(std::move(rawSystemConfig));
```

Comment:
We should create a helper `SystemConfig* jwtSystemConfig();` where we set the correct config. Probably reuse this inside `basicJwtTest`.
We then only use the `changedSystemConfig` to change the JWT config.

Reply:
We should probably have a separate server and client config with override.
See https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/storage_adapters/s3fs/tests/MinioServer.h#L51